modules/textadept/editing.lua

```diff
@@ -461,8 +462,13 @@ function M.enclose(left, right)
     end
     buffer:set_target_range(s, e)
     buffer:replace_target(left .. buffer.target_text .. right)
-    buffer.selection_n_start[i] = buffer.target_end
-    buffer.selection_n_end[i] = buffer.target_end
+    if preserve_sel then
+      buffer.selection_n_start[i] = buffer.target_start + string.len(left)
+      buffer.selection_n_end[i] = buffer.target_end - string.len(right)
```

Comment:
The above `+len(left)` and `-len(right` parts ensure that the just-enclosed `left` and `right` themselves are not included in the selection. Whether this or that mode would be more productive is I guess highly subjective and maybe habitual, but at least the chosen approach resembles the sublime/vscode/etc set. One could offer both but I didn't want to initially unduly bulk up the PR or .lua with "another-new-flags-enum"...  =)